src/sql/inspect.py

```diff
@@ -77,6 +77,51 @@ def _get_row_with_most_keys(rows):
     return list(rows[max_idx])
 
 
+def _is_numeric(value):
+    """Check if a column has numeric and not categorical datatype"""
+    try:
+        float(value)  # Try to convert the value to float
+        return True
+    except ValueError:
+        return False
+
+
+def _check_wrong_datatype(column, value):
+    """Check if a column contains numerical data stored as `str`"""
+    try:
+        if isinstance(value, str) and _is_numeric(value):
+            return True
+        return False
+    except ValueError:
+        pass
+
+
+def _generate_column_styles(
+    column_indices, background_color="#FFFF00", text_color="black"
+):
+    """Change the background-color of all columns with data-type mismatch"""
+    styles = ""
+    for index in column_indices:
+        styles += f"""
+        #profile-table td:nth-child({index + 1}) {{
+            background-color: {background_color};
+            color: {text_color};
+        }}
+        """
+    return f"<style>{styles}</style>"
+
+
+def _generate_message(column_indices, columns):
+    """Generate a message indicating all columns with a datatype mismatch"""
+    message = "Columns "
+    for c in column_indices:
+        col = columns[c - 1]
+        message += f"`{col}`"
```

Comment:
Please use format strings instead of `+`

Reply:
making the changes!